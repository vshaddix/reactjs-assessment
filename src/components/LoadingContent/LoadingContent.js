import React from "react";

export default ({ linesCount = 1 }) => {
  const lines = [];

  for(let x = 0; x < linesCount; x++) {
    lines.push(<div className={"line"}/>);
  }

  return (
    <div className="ui placeholder">
      {lines}
    </div>
  );
};
