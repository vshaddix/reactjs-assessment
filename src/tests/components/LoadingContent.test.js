import React from 'react';
import { create } from "react-test-renderer";
import LoadingContent from '../../components/LoadingContent/LoadingContent';

it('renders (snapshot test)', () => {
  const component = create(<LoadingContent />);

  expect(component.toJSON()).toMatchSnapshot();
});

it('renders the component correctly when passed N lines as argument', () => {
  const component = create(<LoadingContent lines={5} />);

  expect(component.toJSON()).toMatchSnapshot();
});
