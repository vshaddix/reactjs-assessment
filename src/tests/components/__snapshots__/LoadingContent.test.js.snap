// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders (snapshot test) 1`] = `
<div
  className="ui placeholder"
>
  <div
    className="line"
  />
</div>
`;

exports[`renders the component correctly when passed N lines as argument 1`] = `
<div
  className="ui placeholder"
>
  <div
    className="line"
  />
</div>
`;
